ecommerce/frontend/src/components/EmailVerification.js

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react'
+import { api } from '../api'
+import { notification } from 'antd'
+import { sleep } from '../utils'
+import './EmailVerification.less'
+import { CheckCircleTwoTone, CloseCircleTwoTone } from '@ant-design/icons'
+
+const VerificationMessage = verificationState => {
+    console.log(verificationState)
+    if (verificationState == 'Success') {
+        return (
+            <div className="verify-message">
+                <p>Your account has been verified.</p>
+                <CheckCircleTwoTone />
+            </div>
+        )
+    } else if (verificationState == 'Invalid') {
+        return (
+            <div className="verify-message">
+                <p>Failed to verify account</p>
+                <CloseCircleTwoTone twoToneColor="red" />
+            </div>
+        )
+    } else if (verificationState == 'Expired') {
+        return (
+            <div className="verify-message">
+                <p>Your activitation link is expired.</p>
+                <CloseCircleTwoTone twoToneColor="red" />
+            </div>
+        )
+    } else if (verificationState == 'Verified') {
+        return (
+            <div className="verify-message">
+                <p>Your account is already been verified. </p>
+                <CloseCircleTwoTone twoToneColor="red" />
+            </div>
+        )
+    } else {
+        return (
+            <div className="verify-message">
+                <p>Your account is being verified. Please wait...</p>
+            </div>
+        )
+    }
+}
+
+export const EmailVerification = props => {
+    const { id } = props.match.params
+    const [verificationState, setVerificationState] = useState('waiting')
+
+    useEffect(() => {
+        async function fetch() {
+            try {
+                console.log('token: ', id)
+                const { data } = await api.get(`/email-verify/${id}`)
+                console.log(data)
+                await sleep(1000)
+                setVerificationState(data.message)
+            } catch (error) {
+                console.error('failed to verify account', error)
+                setVerificationState('Invalid')
+            } finally {
+            }
+        }
+        fetch()
+    }, [])
+
+    return <div className="verify-message-box">{VerificationMessage(verificationState)}</div>
+}
```

Comment:
nitpick: would be better to use JSX notation for component as in: `<VerificationMessage ... />`